django/db/models/sql/compiler.py

```diff
@@ -150,7 +147,17 @@ def collapse_group_by(self, expressions, having):
                         getattr(expr.output_field, 'model') == self.query.model):
                     pk = expr
             if pk:
+                # MySQLism: Columns in HAVING clause must be added to the GROUP BY.
                 expressions = [pk] + [expr for expr in expressions if expr in having]
+        elif self.connection.features.allows_group_by_selected_pks:
+            # Filter out all expressions associated with a table's primary key present in the grouped columns.
```

Comment:
We usually wrap comments at 79 chars, LGTM otherwise if the ORM guys are happy with it.